examples/basic/gpu_shared.c

```diff
@@ -0,0 +1,191 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ze_api.h>
+
+#include "umf/memory_pool.h"
+#include "umf/pools/pool_disjoint.h"
+#include "umf/providers/provider_level_zero.h"
+
+// This function initializes the Level Zero driver and creates a context in the
+// first GPU device found.
+int level_zero_init(ze_driver_handle_t *driver, ze_device_handle_t *device,
```

Comment:
please add correct error handling in this function
replace asserts with meaningful prints to stderr. 
users will run those examples - sometimes on wrong/miss configured hardware and software - it is really important to tell them what is wrong. Meaningless assert will not help them.

Reply:
done